MyVar/MyVar/Var.cpp

```diff
@@ -0,0 +1,256 @@
+#include "pch.h"
+#include "Var.h"
+#include<iostream>
+using namespace std;
+//constructors
+Var::Var() {
+	i = value_unknow;
+}
+Var::Var(int int_value_) {
+	this->int_value = int_value_;
+	i = value_int;
+}
+Var::Var(double double_value_) {
+	this->double_value = double_value_;
+	i = value_double;
+}
+Var::Var(bool bool_value_) {
+	this->bool_value = bool_value_;
+	i = value_bool;
+}
+Var::Var(char char_value_) {
+	this->char_value = char_value_;
+	i = value_char;
+}
+
+//destructor
+Var::~Var() = default;
+
+//leading to type int 
+Var::operator int() const {
+	try {
+		if (i == value_int) {
```

Comment:
You may use `if` operator for type cheking and raise exception when types are not match.